onnxruntime/python/tools/transformers/notebooks/Tensorflow_Tf2onnx_Bert-Squad_OnnxRuntime_CPU.ipynb

```diff
@@ -316,75 +317,85 @@
+     "text": [
+      "ONNX Runtime cpu inference time for sequence length 512 (model not optimized): 452.52 ms\n"
+     ]
+    }
+   ],
+   "metadata": {}
   },
   {
    "cell_type": "code",
    "execution_count": 11,
-   "metadata": {},
-   "outputs": [],
    "source": [
     "# Some weights of TFBertForQuestionAnswering might not be initialized without fine-tuning.\n",
     "if is_fine_tuned:\n",
     "    print(\"***** Verifying correctness (TensorFlow and ONNX Runtime) *****\")\n",
     "    print('start_scores are close:', numpy.allclose(results[0], start_scores.cpu(), rtol=1e-05, atol=1e-04))\n",
     "    print('end_scores are close:', numpy.allclose(results[1], end_scores.cpu(), rtol=1e-05, atol=1e-04))"
-   ]
+   ],
+   "outputs": [],
+   "metadata": {}
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
    "source": [
     "## 5. Model Optimization\n",
     "\n",
     "[ONNX Runtime BERT Model Optimization Tools](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/transformers) is a set of tools for optimizing and testing BERT models. Let's try some of them on the exported models."
-   ]
+   ],
+   "metadata": {}
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
    "source": [
     "### BERT Optimization Script\n",
     "\n",
-    "The script **optimizer.py** can help optimize BERT model exported by PyTorch, tf2onnx or keras2onnx. Since our model is exported by keras2onnx, we shall use **--model_type bert_keras** parameter.\n",
+    "The script **optimizer.py** can help optimize BERT model exported by PyTorch, tf2onnx or keras2onnx. Since our model is exported by tf2onnx, we shall use **--model_type bert_tf** parameter.\n",
     "\n",
     "It will also tell whether the model is fully optimized or not. If not, that means you might need change the script to fuse some new pattern of subgraph."
-   ]
+   ],
+   "metadata": {}
   },
   {
    "cell_type": "code",
    "execution_count": 12,
-   "metadata": {},
-   "outputs": [],
    "source": [
-    "optimized_model_path =  os.path.join(output_dir, 'keras_{}_opt_cpu.onnx'.format(model_name_or_path))\n",
+    "!{sys.executable} -m pip install --quiet coloredlogs sympy \n",
+    "\n",
+    "optimized_model_path =  os.path.join(output_dir, 'tf2onnx_{}_opt_cpu.onnx'.format(model_name_or_path))\n",
     "\n",
-    "from onnxruntime_tools import optimizer\n",
-    "optimized_model = optimizer.optimize_model(output_model_path, model_type='bert_keras', num_heads=12, hidden_size=768)\n",
+    "from onnxruntime.transformers import optimizer\n",
+    "optimized_model = optimizer.optimize_model(output_model_path, model_type='bert_tf', num_heads=12, hidden_size=768)\n",
     "optimized_model.use_dynamic_axes()\n",
     "optimized_model.save_model_to_file(optimized_model_path)"
-   ]
+   ],
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "failed in shape inference <class 'AttributeError'>\n"
```

Comment:
Failed here. May need take a look.

Reply:
"GlobalAveragepool" and "Reciprocal" is currently not supported in shape inference tool

Reply:
it seems there are more issues besides this two unsupported op.